web-console/knikubevirt/snapshots/snapshots.md

```diff
@@ -0,0 +1,76 @@
+# Virtual Machines Snapshots
+
+Snapshots allow users to create a copy of the VM disks and memory, mainly for restore and backup purposes. The snapshots are taken live or offline depending on if the VM is running or not. 
+This document describes the snapshot design flow.  
+
+
+
+
+## From VM List 
+
+![Offline VM list - snapshot action from a single VM kebab](img/1-00.jpg)
+When clicking on the kebab menu in the VM list, the user will be able to take a snapshot of the VM, meaning he will copy the disks, memory content, and metadata for restoring, backup, creating templates and cloning VMs.
+
+![Offline Snapshot modal - default](img/1-01.jpg)
+In the basic modal, the snapshot will take all current configuration and selected parameters.
+The user must add a descriptive name.
+![Offline Snapshot modal - selecting disks](img/1-02.jpg)
+In the disks field, the user will be able to decide if to include all disks or just some of them and if to include memory as part of the snapshot or not.
+![Offline Snapshot modal - disks customization](img/1-03.jpg)
+![Offline Snapshot modal - disks customized](img/1-04.jpg)
+
+![Offline Snapshotting status indication](img/1-05.jpg)
+![Offline Snapshotting status indication on click and kebab mode](img/1-06.jpg)
+Any virtual machine that has a snapshot in progress will show this in the state:
+When in this progress mode all actions would be disabled and the “Stop snapshot” would be available.
+
+![Offline Restoring from snapshot status](img/1-07.jpg)
+Any virtual machine that is undergoing a restoration process will show a "Restoring" indication on the status, with more details and a "Stop restoring" action on a popover. 
+
+
+## VM page - snapshots tab
+
+![Offline Shanpshot tab](img/2-00.jpg)
+The snapshot list is located under the VM. A tab item in the horizontal navigation will be dedicated for snapshots. The snapshot list is ordered by  date as a default, making the "Scheduled" snapshots appear firts.
+The user can sort the list by name or date, clicking on the title will allow the user to switch sorting and decide if it's ascending or descending.  
+![Offline Snapshot - same modal as in VM list](img/2-01.jpg)
+Same modal as in VM list - Snapshot action
+![Offline Snapshot tab -  snapshot item details](img/2-02a.jpg)
+The user clicks the snapshot's name to view more information. 
+![Offline Snapshot tab - snapshot item details](img/2-02.jpg)
```

Comment:
Typically items in a table like this would be objects in their own right, with their own details pages we could link to. Thus having details in a modal is not something that we have done in the console. Are these snapshots part of the VM object, or is there a VMSnapshot object? 

As a possible  alternative to the modal, we could consider using the side panel to show details. And I can't remember off hand if we use expandable list anywhere in the console, but that could be an option too.

Reply:
I believe the nodes list uses expandable rows, so that might work here as well

Reply:
@cshinn was the expandable rows on nodes list used on older versions?
This is from the Cluster-bot:
<img width="1679" alt="Screen Shot 2019-06-26 at 11 23 24" src="https://user-images.githubusercontent.com/48942365/60163848-1ff63b80-9805-11e9-9794-b2a0af8d0207.png">

Reply:
@Ranelim can we follow up on this, it feels like a gap that will block this PR from being merged.

Reply:
@matthewcarleton 
After some discussions with Tomas, we agree that it makes sense to view the Snapshot details within a Details page since a snapshot IS a Frozen VM. However, it is very important to highlight the hierarchy between the snapshot and its parent/current VM (and the fact that it is a snapshot).

I mentioned this to Moran. He said that we don't have any back-end to support this yet and he is not sure that it would make it into priority any time soon, so he thinks it's kind of a waste for now.
Still, this is just us aligning our goals so...
Sharing two different options to resolve this:
A) Snapshot as a resource
B) Snapshot as a modal

Both start via this link:
https://marvelapp.com/4bfbcf2

Looking at it though mockup, I think the Snapshot details page makes sense in terms of UX. I’m also not sure what is the meta-data we need to float to the user when she chooses to restore the VM within the confirmation modal. I guess it should be a summary of what has changed like Github, or what will be lost. Option A hints that approach in the Restore confirmation modal, but it is not fully resolved, just a notion.

Anyway, I think these changes go beyond the scope of this PR, but if we want to, we can explore it here, or just go with the modal for now.

Reply:
I created an [issue](https://github.com/openshift/openshift-origin-design/issues/227) to follow this discussion, which probably is out of scope for this PR.
@beanh66 WDYT? can I close this conversation?